### PR TITLE
From Aura to Babe

### PR DESCRIPTION
**Pull request by ndkazu**
The 20min Epoch required the block production to use Babe instead of Aura in the runtime.
pallet-share_distributor files & folders were also added

### CODE REVIEW

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -102,11 +127,12 @@ pub fn local_testnet_config() -> Result<ChainSpec, String> {
 			testnet_genesis(
 				wasm_binary,
 				// Initial PoA authorities
```
Comment (ilhanu) [inline node/src/chain_spec.rs:129]:
```suggestion
				// Initial authorities
```

File: node/src/chain_spec.rs [hunk 2]
```diff
@@ -102,11 +127,12 @@ pub fn local_testnet_config() -> Result<ChainSpec, String> {
 			testnet_genesis(
```
Comment (ilhanu) [inline node/src/chain_spec.rs:127]: Why is testnet_genesis in there two times?

File: pallets/onboarding/src/lib.rs [hunk 3]
```diff
@@ -95,8 +95,11 @@ pub mod pallet {
 		type Prop: Parameter
 			+ Dispatchable<Origin = <Self as frame_system::Config>::Origin>
 			+ From<Call<Self>>;
+		#[pallet::constant]
 		type ProposalFee: Get<BalanceOf<Self>>;
 		type WeightInfo: WeightInfo;
```
Comment (ilhanu) [inline pallets/onboarding/src/lib.rs:100]: I understand proposalFee changing in a constant. 
But why is WeightInfo also under the pallet:constant

File: node/src/chain_spec.rs [hunk 4]
```diff
@@ -186,5 +269,37 @@ fn testnet_genesis(
...
+			min_join_bond: 1 * DOLLARS,
+			..Default::default()
+		},
+		staking: StakingConfig {
+			validator_count: initial_authorities.len() as u32,
+			minimum_validator_count: initial_authorities.len() as u32,
+			invulnerables: initial_authorities.iter().map(|x| x.0.clone()).collect(),
+			slash_reward_fraction: Perbill::from_percent(10),
+			stakers,
+			..Default::default()
```
[the remark this answers is not in the document]
Comment (ilhanu) [reply node/src/chain_spec.rs:301]: Also if staking is enabled where exactly is the staking rewards implemented? Usually these are per era

**Review by ilhanu — APPROVED** [review]
As I mentioned in my comments in Discord, this implementation goes above and beyond out of scope for issue #125.  The code checks out and it's future-proof, but we need to be sure we have some follow-up issues to deal with staking, where a PoA was much more simpler understood. 

Please see the review and resolve what you can.

**Review by letodunc — APPROVED** [review]

File: pallets/onboarding/src/lib.rs (hunk 3 above)
Comment (ndkazu) [reply pallets/onboarding/src/lib.rs:100]: #[pallet constant] macro will only affect the type/variable right below it, so WeightInfo is not affected

File: node/src/chain_spec.rs (hunk 2 above)
Comment (ndkazu) [reply node/src/chain_spec.rs:127]: One testnet_genesis used for the `development_config` function, and another one is used for the `local_testnet_config` function

File: node/src/chain_spec.rs [hunk 5]
```diff
@@ -58,11 +82,12 @@ pub fn development_config() -> Result<ChainSpec, String> {
 			testnet_genesis(
 				wasm_binary,
 				// Initial PoA authorities
-				vec![authority_keys_from_seed("Alice")],
+				vec![authority_keys_from_seed_2("Alice")],
```
[the remark this answers is not in the document]
Comment (ndkazu) [reply node/src/chain_spec.rs:85]: Yes, I needed a function slightly different from the original one used by Aura, and its name became `authority_keys_from_seed_2`

File: node/src/chain_spec.rs (hunk 2 above)
Comment (ilhanu) [reply node/src/chain_spec.rs:127]: Ok I missed that in the comparing view.

File: node/src/chain_spec.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (ilhanu) [reply node/src/chain_spec.rs:85]: Can you set it back now Aura is removed?

**Pull request merged**